beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -332,53 +353,159 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
-                }
-                AvailabilityProcessingStatus::MissingComponents(_, _) => {
-                    debug!(
-                        self.log,
-                        "Missing components over rpc";
-                        "block_hash" => %block_root,
-                    );
-                    // Attempt reconstruction here before notifying sync, to avoid sending out more requests
-                    // that we may no longer need.
-                    if let Some(availability) =
-                        self.attempt_data_column_reconstruction(block_root).await
-                    {
-                        result = Ok(availability)
-                    }
-                }
-            },
-            Err(BlockError::DuplicateFullyImported(_)) => {
+        if let &Ok(AvailabilityProcessingStatus::MissingComponents { .. }) = &result {
+            // Attempt reconstruction here before notifying sync, to avoid sending out more requests
+            // that we may no longer need.
+            if let Some(availability) = self.attempt_data_column_reconstruction(block_root).await {
+                result = Ok(availability)
+            }
+        }
+
+        self.send_sync_message(SyncMessage::BlockComponentProcessed {
+            process_type,
+            result: self
+                .handle_lookup_sync_processing_result(block_root, result)
+                .await,
+        });
+    }
+
+    async fn handle_lookup_sync_processing_result(
+        &self,
+        block_root: Hash256,
+        result: Result<AvailabilityProcessingStatus, BlockError>,
+    ) -> LookupSyncProcessingResult {
+        match result {
+            Ok(AvailabilityProcessingStatus::Imported { .. }) => {
                 debug!(
                     self.log,
-                    "Custody columns have already been imported";
-                    "block_hash" => %block_root,
+                    "Fully imported block components over RPC";
+                    "block_root" => %block_root,
                 );
+                self.chain.recompute_head_at_current_slot().await;
+                LookupSyncProcessingResult::FullyImported
             }
-            Err(e) => {
-                warn!(
+            Ok(AvailabilityProcessingStatus::MissingComponents { .. }) => {
+                debug!(
                     self.log,
-                    "Error when importing rpc custody columns";
-                    "error" => ?e,
-                    "block_hash" => %block_root,
+                    "Missing components over RPC";
+                    "block_root" => %block_root,
                 );
+                LookupSyncProcessingResult::ImportedMissingComponents
+            }
+            Err(err) => {
+                debug!(
+                    self.log,
+                    "Error processing block component";
+                    "block_root" => %block_root,
+                    "err" => ?err,
+                );
```

Comment:
Do we want to log internal errors at a higher level than debug? If yes I can move the log statement below and match the type of ErrorCategory